File: 2000 98 & ME/QuickStart/howto/samples/Services/EventLog/logwrite/logwrite.cs
using System;
using System.Diagnostics;

public class WriteLog {
    public static void Main(String[] args) {
        string appName = Environment.GetCommandLineArgs()[0];

        if ( args.Length != 3 ) {
            Console.WriteLine("Usage: " + appName +" <log> <message> <source>");
            return;
        }

        string log = args[0];
        string source = args[2];

        if ( !EventLog.SourceExists(source) ) {
            EventLog.CreateEventSource(source,log);
        }

        EventLog aLog = new EventLog();
        aLog.Source = source;

        if ( aLog.Log.ToUpper() != log.ToUpper() ) {
            Console.WriteLine("Some other application is using the source!");
            return;
        }

        aLog.WriteEntry(args[1],EventLogEntryType.Information);
        Console.WriteLine("Entry written successfuly!");
    }
}

File: 2000 98 & ME/QuickStart/howto/samples/Services/EventLog/logwrite/vb/LogWrite.vb
Imports System
Imports System.Diagnostics

Namespace LogWrite

    Module LogWrite

        Public Sub Main()

            Dim args As String()
            Dim appName As String
            args = Environment.GetCommandLineArgs()
            appName = args(0)

            If (args.Length <> 4) Then
                Console.WriteLine("Usage: " + appName + " <log> <message> <source>")
                Exit Sub
            End If

            Dim log As String
            Dim source As String

            log = args(1)
            source = args(3)

            If (Not EventLog.SourceExists(source)) Then
                EventLog.CreateEventSource(source, log)
            End If

            Dim aLog As EventLog
            aLog = New EventLog
            aLog.Source = source

            If (aLog.Log.ToUpper() <> log.ToUpper()) Then
                Console.WriteLine("Some other application is using the source!")
                Exit Sub
            End If

            aLog.WriteEntry(args(2), EventLogEntryType.Information)
            Console.WriteLine("Entry written successfuly!")
        End Sub

    End Module

End Namespace
